packages/test/test-end-to-end-tests/src/test/summarizerWithLocalChanges.spec.ts

```diff
@@ -0,0 +1,181 @@
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import { IContainerRuntimeOptions, ISummarizer } from "@fluidframework/container-runtime";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { SharedMatrix } from "@fluidframework/matrix";
+import { SharedMap } from "@fluidframework/map";
+import {
+	ITestObjectProvider,
+	waitForContainerConnection,
+	summarizeNow,
+	createSummarizerFromFactory,
+} from "@fluidframework/test-utils";
+import {
+	describeNoCompat,
+	getContainerRuntimeApi,
+	itExpects,
+} from "@fluid-internal/test-version-utils";
+import { IContainerRuntimeBase, IFluidDataStoreFactory } from "@fluidframework/runtime-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { FluidDataStoreRuntime, mixinSummaryHandler } from "@fluidframework/datastore";
+import { pkgVersion } from "../packageVersion";
+
+const runtimeOptions: IContainerRuntimeOptions = {
+	summaryOptions: {
+		summaryConfigOverrides: { state: "disabled" },
+	},
+	gcOptions: { gcAllowed: true },
+};
+export const rootDataObjectType = "@fluid-example/rootDataObject";
+export const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+export const TestDataObjectType2 = "@fluid-example/test-dataStore2";
+class TestDataObject2 extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+	public get _context() {
+		return this.context;
+	}
+	// If this datastore created DDSes it would be even more nodes created without running gc
+}
+
+class TestDataObject1 extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+
+	public get _context() {
+		return this.context;
+	}
+
+	private readonly datastoreKey = "TestDataObject2";
+
+	protected async hasInitialized() {
+		// This can be fired synchronously as well, which would cause an even worse half state
+		await this.init();
+	}
+
+	private async init() {
+		const dataObject2 = await rootDataObjectFactory.createInstance(
+			this.context.containerRuntime,
+		);
+		this.root.set(this.datastoreKey, dataObject2.handle);
+	}
+}
+
+class RootTestDataObject extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+	public get containerRuntime() {
+		return this.context.containerRuntime;
+	}
+}
+
+// Search does something similar to this, where it loads the data object.
+const getComponent = async (runtime: FluidDataStoreRuntime) => {
+	await DataObject.getDataObject(runtime);
+	return undefined;
+};
+
+const rootDataObjectFactory = new DataObjectFactory(
+	rootDataObjectType,
+	RootTestDataObject,
+	[],
+	[],
+	[],
+);
+const dataStoreFactory1 = new DataObjectFactory(
+	TestDataObjectType1,
+	TestDataObject1,
+	[SharedMap.getFactory(), SharedMatrix.getFactory()],
+	[],
+	[],
+	// it would be nice to move away from this flow
```

Comment:
>// it would be nice to move away from this flow [](http://example.com/codeflow?start=1&length=47)

why ? add more info ?

Reply:
Yeah, I would rather not have this comment.

Reply:
OK, I'll remove this